src/scraper/scraper/scraper.py

```diff
@@ -1,31 +1,52 @@
+import codecs
 from functools import partial
 from itertools import ifilter, imap, starmap
 import datetime
 import re
 import os
+import socket
 import ssl
-import urllib2
-
+import logging
+import requests
+import urlparse
 import jsonpickle
 
 from .reddit import reddit_link_elements_generator
 from common.scraperstructs import Document
 
 def _is_valid_link(link_element, invalid_paths_regex, invalid_extensions):
-    path = link_element.url
+    url = link_element.url
+    path = urlparse.urlparse(url).path
     extension = os.path.splitext(path)[1]  # splitext split in two : path and extension
     if extension in invalid_extensions:
         return False
-    invalid_path_found = invalid_paths_regex.search(path)
+    invalid_path_found = invalid_paths_regex.search(url)
     if invalid_path_found:
         return False
     return True
 
+
 def _try_get_html(url):
     try:
-        return urllib2.urlopen(url).read()
-    except (urllib2.URLError, ssl.SSLError): # when login/certificate is required to connect
+        data = requests.get(url, timeout=1)
+        header_encoding = data.encoding
+        if not header_encoding:
+            return None
+        data.content.decode(header_encoding.lower())  # check we can get unicode object from raw data (could raise exception)
+        guessed_encoding = data.apparent_encoding  # important to cache data.apparent_encoding only once (costly)
+        if guessed_encoding is None or header_encoding.lower() != guessed_encoding.lower():
+            return None
+        return data.text.encode('utf-8').decode('utf-8')  # get unicode object in utf-8 format
```

Comment:
https://en.wikipedia.org/wiki/Cargo_cult_programming

it seems to work fine without it and I will remove it.

For the story, what I understood : data.text is an unicode object but the internal byte format is "data.encoding", not necessarily utf-8. The call to "encode" get me a raw byte (str) object, and "decode" will wrap it back to an unicode object, but with an utf-8 internal format.